packages/marshal/src/helpers/promise.js

```diff
@@ -22,27 +27,40 @@ export const PromiseHelper = harden({
   styleName: 'promise',
 
   canBeValid: (candidate, check) => {
+    const passStyleDesc = getOwnPropertyDescriptor(candidate, PASS_STYLE);
     return (
-      (candidate[PASS_STYLE] === 'promise' ||
+      ((passStyleDesc && passStyleDesc.value === 'promise') ||
       isPromise(candidate) ||
-      checkTagRecord(candidate, 'promise', check))
+      check(
+        false,
+        X`Pseudo-promise must be an object with ${q(PASS_STYLE)} ${q(
+          'promise',
+        )}: ${candidate}`,
+      ))
     );
   },
 
   assertValid: candidate => {
-    PromiseHelper.canBeValid(candidate, assertChecker);
     if (isPromise(candidate)) {
       assertSafePromise(candidate);
       return;
     }
 
-    const proto = getPrototypeOf(candidate);
-    proto === objectPrototype ||
-      proto === null ||
+    checkTagRecord(candidate, 'promise', assertChecker);
```

Comment:
Yes, good choice moving `checkTagRecord` from `canBeValid` to `assertValid`.

No change suggested.